va/va.h

```diff
@@ -4791,6 +4791,56 @@ typedef struct _VAPictureHEVC
  */
 #define VA_PICTURE_HEVC_RPS_LT_CURR             0x00000040
 
+typedef enum{
+    VACopyObjectSurface = 0,
+    VACopyObjectBuffer  = 1,
+} VACopyObjectType;
+
+typedef struct _VACopyObject {
+    VACopyObjectType  obj_type;    // type of object.
+    union
+    {
+        VASurfaceID surface_id;
+        VABufferID  buffer_id;
+    } object;
+
+    uint32_t    va_reserved[VA_PADDING_MEDIUM];
+} VACopyObject;
+
+#define VA_COPY_SYNC_BLOCK    0x0
+#define VA_COPY_SYNC_NONBLOCK 0x1
+
+#define VA_COPY_MODE_BALANCE      0x0
+#define VA_COPY_MODE_PERF         0x1
+#define VA_COPY_MODE_POWER_SAVING 0x2
+
+typedef union _VACopyOption{
+    struct {
+        /** \brief va copy synchronization, the value should be VA_COPY_SYNC_XXX
+         * if it equal VA_COPY_SYNC_NONBLOCK , dst surface should be used for sync operation
+         */
+        uint32_t va_copy_sync : 2;
+        /** \brief va copy mode, the value should be VA_COPY_MODE_XXX */
+        uint32_t va_copy_mode : 4;
+        uint32_t reserved     :26;
+    }bits;
+    uint32_t value;
+}VACopyOption;
```

Comment:
We've also planned to expose vaMapBuffer2 API w/ flags and maybe w/ copy options... See #377 for the early draft.

Can we, please, consider whether we can apply `VACopyOption` to the vaMapBuffer2 API as well? I worry about va_copy_sync usage... Thought it might still be useful in vase of vaUnmap when we might or might not to wait for the sync...

Reply:
>I worry about va_copy_sync usage... Thought it might still be useful in vase of vaUnmap when we might or might not to wait for the sync... 

@dvrogozh , do you mean asynchronous writing to vaSurface?

Reply:
Possible usages:
```
vaMapBuffer2(VA_COPY_SYNC_NONBLOCK); // allocates (maybe) shadow copy, no
                              // copy from original surface due to VA_COPY_SYNC_NONBLOCK (we don't care about the contetn)
memcpy();
unmap(); // schedules copy operation form shadow copy to final surface, we don't wait due to VA_COPY_SYNC_NONBLOCK
```

Reply:
Yes, it's async writing to vaSurface/vaBuffer. Above, you say "_usages"_. To tell you the truth I don't see other examples. 
But anyway, IMO, it indeed can be useful.

Reply:
it means map is block call , but unmap is un-block call , right?  the interface will be a little weird. need to re-consider it. maybe also a unmap2?